spec/models/comment_spec.rb

```diff
@@ -0,0 +1,55 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  let(:user) do
+    User.new(
+      name: 'Mateo Lane',
+      photo: 'https://unsplash.com/photos/mateopic',
+      bio: 'Comedian, Italian, Sylish, and part time youtuber',
+      posts_counter: 6
+    )
+  end
+
+  let(:post) do
+    Post.new(
+      author: user,
+      title: 'My first post',
+      text: 'This is my first post',
+      comments_counter: 1,
+      likes_counter: 2
+    )
+  end
+
+  let(:comment) do
+    Comment.new(
+      author: user,
+      post:,
+      text: 'first comment'
+    )
+  end
+
+  it 'is valid with a text' do
+    comment.text = 'first comment'
+    expect(comment).to be_valid
+  end
+
+  it 'is not valid without a text' do
+    comment.text = nil
+    expect(comment).to_not be_valid
+  end
+
+  it 'is only valid with a user' do
+    comment.author = user
+    expect(comment).to be_valid
+  end
+
+  it 'is valid with a post' do
+    comment.post = post
+    expect(comment).to be_valid
+  end
+
+  it 'updates comments counter after save' do
+    comment.save
+    expect(post.comments_counter).to eq(2)
+  end
+end
```

Comment:
An error has occurred, this and all later migrations canceled:

Table 'comments' has no foreign key for users

![Screenshot from 2022-11-25 08-38-03](https://user-images.githubusercontent.com/79058364/203917224-93099a93-3068-401d-9cb2-93b35cddf732.png)